packages/neo4j/src/index.ts

```diff
@@ -1,224 +1,165 @@
+      })
     },
 
     async getUserByEmail(email) {
-      return await query(
-        `MATCH (u:User { email: $email }) RETURN properties(u) AS u`,
-        {
-          email,
-        },
-        {
-          tx: "read",
-        }
+      return await read(
+        `MATCH (u:User { email: $email }) RETURN properties(u)`,
+        { email }
       )
     },
 
     async getUserByAccount(provider_providerAccountId) {
-      return await query(
-        ` MATCH (u:User)-[:HAS_ACCOUNT]->(a:Account {
-            provider: $provider,
-            providerAccountId: $providerAccountId
-          })
-          RETURN properties(u) AS u`,
-        provider_providerAccountId,
-        {
-          tx: "read",
-        }
+      return await read(
+        `MATCH (u:User)-[:HAS_ACCOUNT]->(a:Account {
+           provider: $provider,
+           providerAccountId: $providerAccountId
+         })
+         RETURN properties(u)`,
+        provider_providerAccountId
       )
     },
 
-    async updateUser({ id, emailVerified, ...userData }) {
-      return await query(
-        ` MATCH (u:User { id: $id })
-          SET
-            u += $userData
-            ${
-              undefined !== emailVerified
-                ? `, u.emailVerified = datetime($emailVerified)`
-                : ``
-            }
-          RETURN properties(u) AS u`,
-        {
-          id,
-          userData,
-          emailVerified,
-        }
+    async updateUser(data) {
+      return await write(
+        `MATCH (u:User { id: $data.id })
+         SET u += $data
+         RETURN properties(u)`,
+        data
       )
     },
 
     async deleteUser(id) {
-      return await query(
-        ` MATCH (u:User { id: $id })
-          WITH u, properties(u) AS properties
-          DETACH DELETE u
-          RETURN properties AS deletedUser`,
-        {
-          id,
-        }
+      return await write(
+        `MATCH (u:User { id: $data.id })
+         WITH u, properties(u) AS properties
+         DETACH DELETE u
+         RETURN properties`,
+        { id }
       )
     },
 
-    async linkAccount({ userId, providerAccountId, provider, ...accountData }) {
-      return await query(
-        ` MATCH (u:User { id: $userId })
-          MERGE (a:Account { 
-            providerAccountId: $providerAccountId, 
-            provider: $provider 
-          }) 
-          ON CREATE SET a.id = $id
-          SET a += $accountData
-          MERGE (u)-[:HAS_ACCOUNT]->(a)
-          RETURN a { .*, userId: u.id } AS a`,
-        {
-          id: uuid(),
-          userId,
-          providerAccountId,
-          provider,
-          accountData,
-        }
+    async linkAccount(data) {
+      const account = { id: uuid(), ...data }
+      await write(
+        `MATCH (u:User { id: $data.userId })
+         MERGE (a:Account {
+           providerAccountId: $data.providerAccountId,
+           provider: $data.provider
+         }) 
+         ON CREATE SET a.id = $data.id
```

Comment:
With this change, userId is set on the Account node. 
I think this should is not necessary, because here we create a relationship between Account and User. And, this relationship is be used in `getUserByAccount`.